numba/tests/test_target_extension.py

```diff
@@ -682,6 +667,76 @@ def dpu_foo_cpu():
         for msg in msgs:
             self.assertIn(msg, str(raises.exception))
 
+    def test_overload_allocation(self):
+        def cast_integer(context, builder, val, fromty, toty):
+            # XXX Shouldn't require this.
+            if toty.bitwidth == fromty.bitwidth:
+                # Just a change of signedness
+                return val
+            elif toty.bitwidth < fromty.bitwidth:
+                # Downcast
+                return builder.trunc(val, context.get_value_type(toty))
+            elif fromty.signed:
+                # Signed upcast
+                return builder.sext(val, context.get_value_type(toty))
+            else:
+                # Unsigned upcast
+                return builder.zext(val, context.get_value_type(toty))
+
+        @intrinsic(target='dpu')
+        def intrin_alloc(typingctx, allocsize, align):
+            """Intrinsic to call into the allocator for Array
+            """
+            def codegen(context, builder, signature, args):
+                [allocsize, align] = args
+
+                # XXX: error are being eaten.
+                #      example: replace the next line with `align_u32 = align`
+                align_u32 = cast_integer(context, builder, align,
+                                         signature.args[1], types.uint32)
+                meminfo = context.nrt.meminfo_alloc_aligned(builder, allocsize,
+                                                            align_u32)
+                return meminfo
+
+            from numba.core.typing import signature
+            mip = types.MemInfoPointer(types.voidptr)  # return untyped pointer
+            sig = signature(mip, allocsize, align)
+            return sig, codegen
+
+        @overload_classmethod(types.Array, '_allocate', target='dpu',
+                              nopython=True)
```

Comment:
```suggestion
                              jit_options={'nopython':True)
```
and the same change throughout this test, whilst accepted I think this should really be an error. This change needs making throughout to prevent objmode fallback on the dpu target when doing function resolution (not that it will fail as the function/target resolution call graph is correct, but should it need to be debugged in future).

Reply:
I changed all the `nopython=True` for all the `overload_*`